.github/workflows/nix.yml

```diff
@@ -10,14 +10,39 @@ on:
 
 jobs:
   build:
+    permissions:
+      contents: write
     runs-on: ubuntu-latest
 
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 2
 
-      - uses: cachix/install-nix-action@v16
+      - uses: DeterminateSystems/nix-installer-action@main
```

Comment:
in general, we always pin to exact hashes rather than branches or tags (though it looks like we hadn't previously with `install-nix-action`)

Reply:
Applied 👍